src/flash-source-buffer.js

```diff
@@ -169,6 +169,12 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
     };
 
     chunkInData();
+
+    // fire loaded metadata on first segment append
+    if (!this.firstAppendHappened) {
```

Comment:
I believe it has already finished appending when I call it. Above we call `chunkInData`, which I believe is blocking until we have push the bytes into the source buffer.